src/PhpParser/Node/Manipulator/NullsafeManipulator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\PhpParser\Node\Manipulator;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\NullsafeMethodCall;
+use PhpParser\Node\Expr\NullsafePropertyFetch;
+use PhpParser\Node\Identifier;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+final class NullsafeManipulator
+{
+    public function processNullSafeExpr(Expr $expr): ?Expr
+    {
+        if ($expr instanceof MethodCall) {
+            return new NullsafeMethodCall($expr->var, $expr->name);
+        }
+
+        if (property_exists($expr, 'var') && property_exists($expr, 'name')) {
```

Comment:
This is way too magical check. AST cannot analyse this (PHPStan, Rector and PHPStorm will fail on changed nodes).

It should be handled by `$node instanceof <specificTypeNode>` instead

Reply:
implemented.